docker-storage-setup.sh

```diff
@@ -111,10 +111,48 @@ get_deferred_deletion_string() {
   [ -z "$major" ] && return 0
   [ -z "$minor" ] && return 0
 
-  # docker 1.9 onwards supports deferred device deletion. Enable it.
-  if [ $major -gt 1 ] || ([ $major -eq 1 ] && [ $minor -ge 9 ]);then
-    echo "--storage-opt dm.use_deferred_deletion=true"
-  fi
+  if should_enable_deferred_deletion; then
+     echo "--storage-opt dm.use_deferred_deletion=true"
+  fi
+}
+
+should_enable_deferred_deletion() {
+   # docker 1.9 onwards supports deferred device deletion. Enable it.
+   if [ $major -gt 1 ] || ([ $major -eq 1 ] && [ $minor -ge 9 ]);then
+      if platform_supports_deferred_deletion; then
```

Comment:
Variables major and minor are local to parent function. How does this check work here?